systemtest/apmservertest/config.go

```diff
@@ -169,6 +172,28 @@ type TailSamplingPolicy struct {
 	SampleRate         float64 `json:"sample_rate"`
 }
 
+// ProfilingConfig holds configuration related to profiling.
+type ProfilingConfig struct {
+	Enabled bool `json:"enabled"`
+
+	// ESConfig holds Elasticsearch configuration for writing
+	// profiling stacktrace events and metadata documents.
+	ESConfig *ElasticsearchOutputConfig `json:"elasticsearch"`
+
+	// MetricsESConfig holds Elasticsearch configuration for
+	// writing profiling host agent metric documents.
+	MetricsESConfig *ElasticsearchOutputConfig `json:"metrics.elasticsearch"`
+
+	// ILMConfig
+	ILMConfig *ProfilingILMConfig `json:"keyvalue_retention"`
+}
+
+type ProfilingILMConfig struct {
+	Age         time.Duration `json:"age"`
+	SizeInBytes uint64        `json:"size_bytes"`
+	Interval    time.Duration `json:"execution_interval"`
+}
```

Comment:
I think we should leave ILMConfig out, and remove the ProfilingILMConfig. It's not used AFAICS. If it were, I think we'd need to add a custom MarshalJSON method like in other structs, for marshalling the time.Duration as expected.

Reply:
Good catch! Updated!